pyca/putfunctions.hh

```diff
@@ -60,13 +60,16 @@ void _pyca_put_value(capv* pv, PyObject* pyvalue, T** buf, long count)
   }
   T* buffer = reinterpret_cast<T*>(pv->putbuffer);
   if (count == 1) {
+      // if we only want to put the first element
       if (PyTuple_Check(pyvalue)) {
         PyObject* pyval = PyTuple_GetItem(pyvalue, 0);
         _pyca_put(pyval, buffer);
       } else if (PyArray_Check(pyvalue)) {
-        void* npdata = PyArray_GETPTR1(pyvalue, 0);
+        // Convert to array
+        PyArrayObject *arr = (PyArrayObject *)PyArray_FROM_O(pyvalue); 
+        char* npdata = static_cast<char*>(PyArray_GETPTR1(arr, 0));
         if (PyArray_IsPythonScalar(pyvalue)) {
-          PyObject* pyval = PyArray_GETITEM(pyvalue, npdata);
+          PyObject* pyval = PyArray_GETITEM(arr, npdata);
```

Comment:
Note for the reviewer:  This change revolves around casting `npdata` to a `char*`, despite the numpy docs insisting that `PyArray_GETITEM` takes a `void*` as its second argument ([api docs](https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_GETITEM)).  

This was, of course, changed in reaction to compilation errors.  These errors could be downgraded to warnings if `-fpermissive` was specified as a compiler arg, but other source code sources seemed to corroborate this slightly more specific signature [numpy/_core/include/numpy/ndarrayobject.h](https://github.com/numpy/numpy/blob/1af7cb3c2d42800a5614baef5b5fdadad808e4c4/numpy/_core/include/numpy/ndarrayobject.h#L279-L284)

Reply:
The linked function then take the `char*` and casts it to a `void*`, but I'm not digging any further into this...